### PR TITLE
feat: add support for Request / Response progress

### PR DESCRIPTION
**Pull request by picunada**
### 🔗 Linked issue

Issue #45 

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Added support for Request / Response progress.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by picunada** [comment]
Hello, would like to request help for this PR.
First commit here was working probably only within Browser env.
How can I add support for Node streams?
Also requesting refactor advices :)

**Comment by pi0** [comment]
Hi dear @picunada thanks for picking on this and nice idea for initial POC.

I think in order to move this idea forward, i am mainly looking forward to exposing minimum enough hooks in order to be able creating `fetchWithProgrsss` utility by wrapping `$fetch`, this way we can both make sure not adding overhead to the core and also implement platform dependent implementations. Also we could directly support a binary transform to allow all kind of responses. Not sure how we can wrap native utils such as `response.formData()` (one idea might be wrapping `Response`)

Basically what am i looking forward for next steps is to somehow introduce something like `TrackedRequest` / `TrackedResponse` wrapper classes. They might make this feature complete

**Comment by Dmytro-Tihunov** [comment]
> Hi dear @picunada thanks for picking on this and nice idea for initial POC.
> 
> 
> 
> I think in order to move this idea forward, i am mainly looking forward to exposing minimum enough hooks in order to be able creating `fetchWithProgrsss` utility by wrapping `$fetch`, this way we can both make sure not adding overhead to the core and also implement platform dependent implementations. Also we could directly support a binary transform to allow all kind of responses. Not sure how we can wrap native utils such as `response.formData()` (one idea might be wrapping `Response`)
> 
> 
> 
> Basically what am i looking forward for next steps is to somehow introduce something like `TrackedRequest` / `TrackedResponse` wrapper classes. They might make this feature complete 

Looking forward to this

**Comment by ChrisGV04** [comment]
Can this PR be expected to be implemented soon? It would be really nice to have on Nuxt 3!
